src/parser/util.js

```diff
@@ -65,7 +65,18 @@ pp.isLineBreak = function () {
 pp.canInsertSemicolon = function () {
   return this.match(tt.eof) ||
     this.match(tt.braceR) ||
-    this.isLineBreak();
+    this.isLineBreak() ||
+    (this.hasPlugin("lightscript") && (
+      // LSC oneline statement ASI cases
+      // Allow if x: throw y else: throw z
+      this.match(tt._else) ||
+      this.match(tt._elif) ||
+      // Allow (-> throw new Error)()
+      this.match(tt.parenR) ||
+      // Technically it is legal to insert a ; after a ;.
+      // Allows -> throw new Error; f()
+      this.state.tokens[this.state.tokens.length - 1].type === tt.semi
```

Comment:
It's not clear to me how that would happen. How is `(-> throw new Error)()` going to compile? It calls `pp.semicolon()` after parsing the throw, but there isn't a semicolon, EOL, or brace there...